### ISSUE TITLE
gitlab_runner: ability to provide authentikation token for runner registration

### ISSUE THREAD

**Issue by DasFaultier** [ansible-collections/community.general#7559]
### Summary

From https://docs.gitlab.com/ee/ci/runners/new_creation_workflow:
> In GitLab 16.0, we introduced a new runner creation workflow that uses runner authentication tokens to register runners. The legacy workflow that uses registration tokens is deprecated and will be removed in GitLab 18.0.

This conflicts with the way that the gitlab_runner module currently works. From its [documentation](https://docs.ansible.com/ansible/latest/collections/community/general/gitlab_runner_module.html#parameter-registration_token):
> The registration token is used to register new runners.
> Required if `[state=present]`

To stay up to date with the latest GitLab versions, the `gitlab_runner` module will probably have to implement a parameter for providing the authentication token that is now used instead of the registration token. This authentication token is obtained by manually creating a Runner in the Gitlab UI.

From what I understand, an alternative approach might be to use a personal access token (`glpat-*`) to register a runner via REST API (`/user/runners`) and use the authentikation token from the JSON response to run any other necessary actions.

I'm by no means an expert with any of this and I've only played around with GitLab's REST API a little, so please take what comes next with a grain of salt. I suspect that in order to make the module idempotent, more effort might be required. Calling the `/user/runners` endpoint will always register a new runner with a fresh ID, so in order to, for example, create a runner with a specific description only once and skip its creation in all subsequent runs, the module would also have to run calls against `groups/{{ item.value.id }}/runners` and similar endpoints to gather information about existing runners and skip the creation step if necessary. I have no idea how much of the necessary functionality if already covered in the `python-gitlab` library.

### Issue Type

Feature Idea

### Component Name

gitlab_runner

### Additional Information

```yaml (paste below)

```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by ansibullbot** [comment]
Files identified in the description:
* [`plugins/modules/gitlab_runner.py`](https://github.com/ansible-collections/community.general/blob/main/plugins/modules/gitlab_runner.py)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/devel/ISSUE_HELP.md)

**Comment by ansibullbot** [comment]
cc @Lunik @SamyCoenen @Shaps @lgatellier @marwatk @metanovii @nejch @scodeman @sh0shin @suukit @waheedi @zanssa
[click here for bot help](https://github.com/ansible/ansibullbot/blob/devel/ISSUE_HELP.md)

**Comment by lgatellier** [comment]
Hi @DasFaultier,

The new runner creation workflow has been addressed in #3742 😉 
This feature will soon be released in version `8.1.0`. It will require `python-gitlab` [`v4.0.0`](https://github.com/python-gitlab/python-gitlab/releases/tag/v4.0.0).

Feel free to raise an issue if your encounter some bugs.

duplicate_of 3742

**Comment by DasFaultier** [comment]
My apologies, for some reason I didn't find that issue before writing my own. Thank you for your work, I'm looking forward to the release!
